src/datasource.ts

```diff
@@ -24,4 +30,53 @@ export class DataSource extends DataSourceWithBackend<AthenaQuery, AthenaDataSou
 
   applyTemplateVariables = (query: AthenaQuery, scopedVars: ScopedVars) =>
     applySQLTemplateVariables(query, scopedVars, getTemplateSrv);
+
+  getVariables = () => this.templateSrv.getVariables().map((v) => `$${v.name}`);
+
+  getRegions = () => this.getResource('regions').then((regions) => appendTemplateVariables(this, regions));
+
+  getCatalogs = (query: AthenaQuery) =>
+    this.postResource('catalogs', {
+      region: this.templateSrv.replace(query.connectionArgs.region),
+    }).then((catalogs) => appendTemplateVariables(this, catalogs));
+
+  getDatabases = (query: AthenaQuery) =>
+    this.postResource('databases', {
+      region: this.templateSrv.replace(query.connectionArgs.region),
+      catalog: this.templateSrv.replace(query.connectionArgs.catalog),
+    }).then((databases) => appendTemplateVariables(this, databases));
+
+  getTables = (query: AthenaQuery) =>
+    this.postResource('tables', {
+      region: this.templateSrv.replace(query.connectionArgs.region),
+      catalog: this.templateSrv.replace(query.connectionArgs.catalog),
+      database: this.templateSrv.replace(query.connectionArgs.database),
+    }).then((tables) => appendTemplateVariables(this, tables));
+
+  getColumns = (query: AthenaQuery) =>
+    this.postResource('columns', {
+      region: this.templateSrv.replace(query.connectionArgs.region),
+      catalog: this.templateSrv.replace(query.connectionArgs.catalog),
+      database: this.templateSrv.replace(query.connectionArgs.database),
+      table: this.templateSrv.replace(query.table),
+    }).then((columns) => appendTemplateVariables(this, columns));
```

Comment:
Datasource is mocked in the QueryEditor tests so these lines are not covered by unit tests (see code cov bot report).
Could you add a test file for this?

Reply:
I'll work on adding tests for `datasource.ts` 😊

Reply:
I think the new tests should provide sufficient coverage (I still need to add a test for the query method itself).